lib/explorer/polars_backend/data_frame.ex

```diff
@@ -280,15 +280,16 @@ defmodule Explorer.PolarsBackend.DataFrame do
   def collect(df), do: df
 
   @impl true
-  def from_tabular(tabular) do
+  def from_tabular(tabular, dtypes) do
     {_, %{columns: keys}, _} = reader = init_reader!(tabular)
     columns = Table.to_columns(reader)
 
     keys
     |> Enum.map(fn key ->
+      dtype = dtypes |> Map.new() |> Map.get(key)
```

Comment:
I think we don't need to create a new map for each key. You could move the `dtypes |> Map.new()` to outside the `Enum.map/2` function.

Alternatively you could use [`List.keyfind/3`](https://hexdocs.pm/elixir/List.html#keyfind/4) instead of creating a map.

Reply:
Yes, it makes sense